esp-wifi/src/wifi/mod.rs

```diff
@@ -1424,6 +1495,19 @@ mod asynch {
             Ok((result, count))
         }
 
+        pub async fn scan_with_config<const N: usize>(
+            &mut self,
+            config: ScanConfig<'_>,
+        ) -> Result<heapless::Vec<AccessPointInfo, N>, WifiError> {
+            Self::clear_events(WifiEvent::ScanDone);
+            esp_wifi_result!(wifi_start_scan(false, config))?;
+            WifiEventFuture::new(WifiEvent::ScanDone).await;
+
+            let result = self.scan_results()?;
```

Comment:
You should stay consistent with the method above and return the count as well. It shows if there are more networks than space in the vector.

Reply:
Changed that